src/MetricExporter.ts

```diff
@@ -96,135 +87,73 @@ export class DynatraceMetricExporter implements MetricExporter {
-			const dimensions = Object.entries(metric.attributes).map(([key, value]) => ({ key, value }));
-			switch (metric.aggregator.kind) {
-				case AggregatorKind.SUM: {
-					if (metric.aggregationTemporality !== AggregationTemporality.AGGREGATION_TEMPORALITY_DELTA) {
-						diag.warn(`Aggregation temporality ${AggregationTemporality[metric.aggregationTemporality]} unsupported for metric ${metric.descriptor.name}`);
-						return null;
-					}
-					const data = metric.aggregator.toPoint();
-
-					const valueType = typeof data.value;
-					if (valueType !== "number") {
-						diag.warn(`Metric ${metric.descriptor.name} has invalid data.value type ${valueType}`);
-						return null;
-					}
-
-					if (!isFinite(data.value)) {
-						diag.warn(`Metric ${metric.descriptor.name} has non-finite value`);
-						return null;
-					}
-
-					const normalizedMetric = this._dtMetricFactory
-						.createCounterDelta(
-							metric.descriptor.name,
-							dimensions,
-							data.value,
-							new Date(hrTimeToMilliseconds(data.timestamp))
-						);
-					if (normalizedMetric == null) {
-						this._warnNormalizationFailure(metric.descriptor.name);
-					}
-					return normalizedMetric;
-				}
-				case AggregatorKind.HISTOGRAM: {
-					const data = metric.aggregator.toPoint();
-					const { sum, count } = data.value;
-
-					// TODO remove this hack which pretends all data points had the same value
-					const avg = sum / count;
-
-					const value: SummaryValue = {
-						min: avg,
-						max: avg,
-						sum,
-						count
-					};
-					const normalizedMetric = this._dtMetricFactory
-						.createSummary(
-							metric.descriptor.name,
-							dimensions,
-							value,
-							new Date(hrTimeToMilliseconds(data.timestamp))
-						);
-					if (normalizedMetric == null) {
-						this._warnNormalizationFailure(metric.descriptor.name);
-					}
-					return normalizedMetric;
-				}
-				case AggregatorKind.LAST_VALUE: {
-					const data = metric.aggregator.toPoint();
-					const valueType = typeof data.value;
-					if (valueType !== "number") {
-						diag.warn(`Metric ${metric.descriptor.name} has invalid data.value type ${valueType}`);
-						return null;
-					}
-
-					if (!isFinite(data.value)) {
-						diag.warn(`Metric ${metric.descriptor.name} has non-finite value`);
-						return null;
-					}
-
-					const normalizedMetric = this._dtMetricFactory
-						.createGauge(
-							metric.descriptor.name,
-							dimensions,
-							data.value,
-							new Date(hrTimeToMilliseconds(data.timestamp))
-						);
-					if (normalizedMetric == null) {
-						this._warnNormalizationFailure(metric.descriptor.name);
-					}
-					return normalizedMetric;
+		const lines: string[] = [];
+		for (const scopeMetric of metrics.scopeMetrics) {
+			for (const metric of scopeMetric.metrics) {
+				switch (metric.descriptor.type) {
+					case InstrumentType.COUNTER:
+					case InstrumentType.OBSERVABLE_COUNTER:
+						lines.push(...this.serializeCounter(metric));
```

Comment:
Could `lines.concat(this.serializeCounter(metric))` faster here? :thinking: 
Not sure about the performance impact of using the spread operator.